core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/BatchCubeSink.java

```diff
@@ -69,24 +75,44 @@ public BatchCubeSink(CubeSinkConfig config) {
   @Override
   public void initialize(BatchRuntimeContext context) throws Exception {
     super.initialize(context);
-    transform = new StructuredRecordToCubeFact(context.getPluginProperties().getProperties());
+    transform = new StructuredRecordToCubeFact(getProperties());
   }
 
+  @VisibleForTesting
   @Override
   protected Map<String, String> getProperties() {
-    Map<String, String> properties = new HashMap<>(config.getProperties().getProperties());
-    // todo: workaround for CDAP-2944: allows specifying custom props via JSON map in a property value
-    if (!Strings.isNullOrEmpty(config.getDatasetOther())) {
-      properties.remove(Properties.Cube.DATASET_OTHER);
-      Map<String, String> customProperties = GSON.fromJson(config.getDatasetOther(), STRING_MAP_TYPE);
-      properties.putAll(customProperties);
+    Map<String, String> properties = new HashMap<>();
+    // done only for testing
+    if (config.getProperties() != null) {
+      properties.putAll(config.getProperties().getProperties());
+    }
+    // add aggregations
+    if (!Strings.isNullOrEmpty(config.getAggregations())) {
+      properties.remove(Properties.Cube.AGGREGATIONS);
+      properties.putAll(CubeUtils.parseAndGetProperties(Properties.Cube.AGGREGATIONS,
+                                                        config.getAggregations(), ";", ":",
+                                                        new Function<String, String>() {
+                                                          @Override
+                                                          public String apply(String input) {
+                                                            return "dataset.cube.aggregation." + input + ".dimensions";
+                                                          }
+                                                        }));
+    }
+
+    // add measurements
+    if (!Strings.isNullOrEmpty(config.getMeasurements())) {
+      properties.remove(Properties.Cube.MEASUREMENTS);
+      properties.putAll(CubeUtils.parseAndGetProperties(Properties.Cube.MEASUREMENTS,
+                                                        config.getMeasurements(), ",", ":",
```

Comment:
Wondering if we can use ';' as the delimiter in both cases? Bigger change I guess?